renamed 'rules' as 'evaluators'

annaouyang opened the pull request:
We decided that the terminology 'rule' was misleading; rather, a better name would be 'evaluator'. Code was refactored to reflect this change. 

Examples of changes: 
- RuleEngine --> EvaluatorEngine
- SentenceRule --> SentenceEvaluator
- applyRule --> applyEvaluator

 All test cases passed.

zappyfish commented on src/main/java/opinion/evaluators/SentenceEvaluator.java:
```diff
@@ -0,0 +1,24 @@
+
+    public abstract EvaluatorResult.EvaluatorType getEvaluatorType();
+
+    public abstract boolean evaluatorApplies(Sentence sentence);
```
The motivation behind this refactor was realizing that many solutions to classifying sentences don't approach the problem from

```
if (ruleApplies(sentence)) {
    // we can make a conclusion i.e. opinion or fact
}
```

but instead:

`classification = evaluateSentence(sentence)`

That is to say, there is no question whether an evaluator applies. An evaluator always returns something, and specifically can return either `opinion` or `fact` (though we should also support `no result`). So while the renaming here is good, we also need to refactor method interfaces a bit.

annaouyang commented:
Some major changes: 

1. Evaluator 'description' and 'explanation' were being used to refer to the same thing --> now everything is called 'description' (e.g., getEvaluatorDescription())

2. Changed the interface for EvaluatorResult: 
- there are no longer enums for "EvaluatorTypes" (formerly RuleTypes); instead we have enums for Evaluations as follows: FACT, OPINION, NO_RESULT
- as such, there is no longer a method for getEvaluatorType(), only getEvaluation() and getEvaluatorDescription()
- since any evaluator can be applied to a sentence, ruleApplies() was also taken out

3. Changed the interface for SentenceEvaluator: 
- due to the changes in EvaluatorResult, SentenceEvaluator needed to be modified as well
- now each child of SentenceEvaluator only overrides 2 methods: getEvaluatorDescription() and getEvaluation(Sentence sentence) --> note that getEvaluation(Sentence sentence) returns OPINION, FACT, or NO_RESULT

** Instead of applying an opinion "rule" to a sentence and returning true or false (e.g., a sentence is an opinion when we have a SentenceRule that is of an opinion type and it returns true when it is applied), we are now applying 'evaluators' to sentences that classify sentences as fact, opinion, or no_result

4. In EvaluatorFactory: 
getRuleTypes() was changed to getEvaluatorNames() --> this method now gets the names of all evaluators
